### PR TITLE
Add table aws_servicecatalog_portfolio. closes #1635

### LINKED ISSUE #1635: Add table aws_servicecatalog_portfolio
**References**
List: https://docs.aws.amazon.com/servicecatalog/latest/dg/API_ListPortfolios.html
Get: https://docs.aws.amazon.com/servicecatalog/latest/dg/API_DescribePortfolio.html

TF: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/servicecatalog_portfolio

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_servicecatalog_portfolio []

PRETEST: tests/aws_servicecatalog_portfolio

TEST: tests/aws_servicecatalog_portfolio
Running terraform
data.aws_region.alternate: Reading...
data.aws_canonical_user_id.current_user: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_region.primary: Reading...
data.aws_partition.current: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-west-2]
data.aws_caller_identity.current: Read complete after 1s [id=123444323243]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]
data.aws_canonical_user_id.current_user: Read complete after 1s [id=d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_servicecatalog_portfolio.portfolio will be created
  + resource "aws_servicecatalog_portfolio" "portfolio" {
      + arn           = (known after apply)
      + created_time  = (known after apply)
      + description   = "List of my organizations apps"
      + id            = (known after apply)
      + name          = "turbottest75419"
      + provider_name = "turbottest75419"
      + tags_all      = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + aws_account   = "123444323243"
  + aws_region    = "us-west-2"
  + resource_aka  = (known after apply)
  + resource_id   = (known after apply)
  + resource_name = "turbottest75419"
aws_servicecatalog_portfolio.portfolio: Creating...
aws_servicecatalog_portfolio.portfolio: Creation complete after 2s [id=port-hgadege5bgb3s]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 44, in data "null_data_source" "resource":
  44: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

aws_account = "123444323243"
aws_region = "us-west-2"
resource_aka = "arn:aws:catalog:us-west-2:123444323243:portfolio/port-hgadege5bgb3s"
resource_id = "port-hgadege5bgb3s"
resource_name = "turbottest75419"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:catalog:us-west-2:123444323243:portfolio/port-hgadege5bgb3s",
    "display_name": "turbottest75419",
    "id": "port-hgadege5bgb3s",
    "provider_name": "turbottest75419"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "display_name": "turbottest75419",
    "id": "port-hgadege5bgb3s"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "123444323243",
    "akas": [
      "arn:aws:catalog:us-west-2:123444323243:portfolio/port-hgadege5bgb3s"
    ],
    "display_name": "turbottest75419",
    "region": "us-west-2",
    "title": "turbottest75419"
  }
]
✔ PASSED

POSTTEST: tests/aws_servicecatalog_portfolio

TEARDOWN: tests/aws_servicecatalog_portfolio

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_servicecatalog_portfolio
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+-------------+-------------------------------------------------------------+---------------->
| display_name   | id                 | arn                                                                 | created_time              | description | provider_name | budgets | tag_options | tags_src                                                    | title          >
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+-------------+-------------------------------------------------------------+---------------->
| test-portfolio | port-p4asc5n56lqrk | arn:aws:catalog:us-east-1:123444323243:portfolio/port-p4asc5n56lqrk | 2023-03-03T16:50:44+05:30 | testing     | test          | []      | []          | [{"Key":"foo","Value":"bar"},{"Key":"foo1","Value":"bar1"}] | test-portfolio >
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+-------------+-------------------------------------------------------------+---------------

> select * from aws_servicecatalog_portfolio where id = 'port-p4asc5n56lqrk'
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+------------+-------------------------------------------------------------+----------------+>
| display_name   | id                 | arn                                                                 | created_time              | description | provider_name | Budgets | TagOptions | tags_src                                                    | title          |>
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+------------+-------------------------------------------------------------+----------------+>
| test-portfolio | port-p4asc5n56lqrk | arn:aws:catalog:us-east-1:123444323243:portfolio/port-p4asc5n56lqrk | 2023-03-03T16:50:44+05:30 | testing     | test          | []      | []         | [{"Key":"foo","Value":"bar"},{"Key":"foo1","Value":"bar1"}] | test-portfolio |>
+----------------+--------------------+---------------------------------------------------------------------+---------------------------+-------------+---------------+---------+------------+-------------------------------------------------------------+----------------
```
</details>

### CODE REVIEW

**Comment by wanisfahmyDE** [comment]
Hi @ParthaI,

Thanks for implementing this PR!

I might be wrong but this would only return the service catalog portfolios which is just a logical grouping for the products. But it won't return the products itself nor the product "instances" created. Or did I misunderstand?

File: aws/table_aws_servicecatalog_portfolio.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+			},
+			{
+				Name:        "tags_src",
+				Description: "Information about the tags associated with the portfolio.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getServiceCatalogPortfolio,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns
```
Comment (misraved) [inline aws/table_aws_servicecatalog_portfolio.go:92]:
```suggestion
			// Steampipe standard columns
```

File: aws/table_aws_servicecatalog_portfolio.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+//// LIST FUNCTION
+
+func listServiceCatalogPortfolios(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create Client
+	svc, err := ServiceCatalogClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_servicecatalog_portfolio.listServiceCatalogPortfolios", "client_error", err)
```
Comment (misraved) [inline aws/table_aws_servicecatalog_portfolio.go:123]:
```suggestion
		plugin.Logger(ctx).Error("aws_servicecatalog_portfolio.listServiceCatalogPortfolios", "connection_error", err)
```

File: aws/table_aws_servicecatalog_portfolio.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	}
+
+	if id == "" {
+		return nil, nil
+	}
+
+	// Create client
+	svc, err := ServiceCatalogClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_servicecatalog_portfolio.getServiceCatalogPortfolio", "client_error", err)
```
Comment (misraved) [inline aws/table_aws_servicecatalog_portfolio.go:191]:
```suggestion
		plugin.Logger(ctx).Error("aws_servicecatalog_portfolio.getServiceCatalogPortfolio", "connection_error", err)
```

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!!

**Review by misraved — APPROVED** [review]

**Pull request merged**